diff --git a/greenreg/src/internal/registers/register_data/primary_register_data.cpp b/greenreg/src/internal/registers/register_data/primary_register_data.cpp
index a1d9d51..0961b41 100644
--- a/greenreg/src/internal/registers/register_data/primary_register_data.cpp
+++ b/greenreg/src/internal/registers/register_data/primary_register_data.cpp
@@ -139,10 +139,10 @@ void primary_register_data::on_received_write_request( uint_gr_t _value, transac
       }
   }
     
-  m_parent_register->i.set( _value);
+  m_parent_register->i.set(new_value);
 
-  m_parent_register->get_post_write_rules().process_active_rules( old_value, _value, _transaction, _delayed);
-  m_parent_register->br.process_tlm_post_write_rules( old_value, _value, _transaction, _delayed);
+  m_parent_register->get_post_write_rules().process_active_rules( old_value, new_value, _transaction, _delayed);
+  m_parent_register->br.process_tlm_post_write_rules( old_value, new_value, _transaction, _delayed);
   
   // potentially call wait, dependent on switch and current context
   if (m_parent_register->get_events_enabled() || _delayed) {
