diff -rupN a/CULT/CMakeLists.txt b/CULT/CMakeLists.txt
--- a/CULT/CMakeLists.txt	2013-01-07 09:25:03.000000000 +0100
+++ b/CULT/CMakeLists.txt	2013-07-18 13:06:38.967647484 +0200
@@ -14,7 +14,7 @@ endif (MSVC)
 
 set(Boost_USE_STATIC_LIBS ON)
 
-find_package (Boost 1.48 COMPONENTS thread date_time system serialization REQUIRED)
+find_package (Boost 1.42 COMPONENTS thread date_time system serialization REQUIRED)
 
 include_directories (${Boost_INCLUDE_DIRS})
 link_directories (${Boost_LIBRARY_DIRS})
diff -rupN a/CULT/include/cult-base.h b/CULT/include/cult-base.h
--- a/CULT/include/cult-base.h	2013-04-04 15:10:46.000000000 +0200
+++ b/CULT/include/cult-base.h	2013-07-18 13:04:13.147070632 +0200
@@ -48,7 +48,7 @@
 #define CULT_VAR_INIT_SC(var, n, l, m) var.cultInitialize(n, l, m);	//initialize log variable with sc_time
 #define CULT_VAR_LOG(v) v.log(true)		//explicit logging
 #define CULT_VAR_SET_MAX(v, max) v.setmax(max)	//set maximum value for filter
-#define CULT_VAR_SET_MIN(v, min) v.setmax(min)	//set minimum value for filter
+#define CULT_VAR_SET_MIN(v, min) v.setmin(min)	//set minimum value for filter
 
 #define CULT_LOG_START cult::Manager::getInstance().startLog()		//start logging
 #define CULT_LOG_STOP cult::Manager::getInstance().stopLog()		//stop logging
@@ -79,7 +79,7 @@
 #define CULT_VAR_INIT_SC(var, n, l, m)
 #define CULT_VAR_LOG(v)
 #define CULT_LOG_MESSAGE(l, t)
-#define CULT_LOG_MESSAGE(m, l, t)
+#define CULT_LOG_MESSAGE_SC(m, l, t)
 #define CULT_LOG_BACKWARDS
 #define CULT_LOG_START
 #define CULT_LOG_STOP
