diff -Nur greensocs-4.0.0/greenreg/internal/registers/register_data/primary_register_data.cpp greensocs-4.0.0_mod/greenreg/internal/registers/register_data/primary_register_data.cpp
--- greensocs-4.0.0/greenreg/internal/registers/register_data/primary_register_data.cpp	2010-02-18 17:56:36.000000000 +0100
+++ greensocs-4.0.0_mod/greenreg/internal/registers/register_data/primary_register_data.cpp	2010-10-01 12:56:37.471482853 +0200
@@ -141,10 +141,10 @@
       }
   }
     
-  m_parent_register->i.set( _value);
+  m_parent_register->i.set(new_value);
 
-  m_parent_register->get_post_write_rules().process_active_rules( old_value, _value, _transaction, _delayed);
-  m_parent_register->br.process_tlm_post_write_rules( old_value, _value, _transaction, _delayed);
+  m_parent_register->get_post_write_rules().process_active_rules( old_value, new_value, _transaction, _delayed);
+  m_parent_register->br.process_tlm_post_write_rules( old_value, new_value, _transaction, _delayed);
   
   // potentially call wait, dependent on switch and current context
   if (m_parent_register->get_events_enabled() || _delayed) {
