--- a/greenscript/lib/CMakeLists.txt	2014-07-20 10:55:06.000000000 +0200
+++ b/greenscript/lib/CMakeLists.txt	2014-07-20 10:57:10.000000000 +0200
@@ -7,21 +7,21 @@
 #
 # Create _gsp_sc.so and _gsp_sc.py for embedded Python scripts.
 #
-SET_SOURCE_FILES_PROPERTIES(../gsp_scPYTHON_wrap.cxx PROPERTIES GENERATED 1)
+SET_SOURCE_FILES_PROPERTIES(../lib/gsp_scPYTHON_wrap.cxx PROPERTIES GENERATED 1)
 SET_SOURCE_FILES_PROPERTIES(../gsp_sc.i PROPERTIES CPLUSPLUS ON)
 SET_SOURCE_FILES_PROPERTIES(../gsp_sc.i
                                      PROPERTIES SWIG_FLAGS "-DUSE_GREENCONTROL")
 SWIG_ADD_MODULE(gsp_sc python ../gsp_sc.i)
 
 ADD_LIBRARY(_gsp_sc.so SHARED ../src/gsp_sc.cpp
-                              ../gsp_scPYTHON_wrap.cxx)
+                              ../lib/gsp_scPYTHON_wrap.cxx)
 SET_TARGET_PROPERTIES(_gsp_sc.so PROPERTIES PREFIX "")
 SET_TARGET_PROPERTIES(_gsp_sc.so PROPERTIES SUFFIX "")
 
 #
 # Create _gsp_sc_sa.so and _gsp_sc_sa.py for standalone Python scripts.
 #
-SET_SOURCE_FILES_PROPERTIES(../gsp_sc_saPYTHON_wrap.cxx
+SET_SOURCE_FILES_PROPERTIES(../lib/gsp_sc_saPYTHON_wrap.cxx
                                                          PROPERTIES GENERATED 1)
 SET_SOURCE_FILES_PROPERTIES(../gsp_sc_sa.i PROPERTIES CPLUSPLUS ON)
 SET_SOURCE_FILES_PROPERTIES(../gsp_sc_sa.i
@@ -29,7 +29,7 @@
 SWIG_ADD_MODULE(gsp_sc_sa python ../gsp_sc_sa.i)
 
 ADD_LIBRARY(_gsp_sc_sa.so SHARED ../src/gsp_sc.cpp
-                                 ../gsp_sc_saPYTHON_wrap.cxx)
+                                 ../lib/gsp_sc_saPYTHON_wrap.cxx)
 SET_TARGET_PROPERTIES(_gsp_sc_sa.so PROPERTIES PREFIX "")
 SET_TARGET_PROPERTIES(_gsp_sc_sa.so PROPERTIES SUFFIX "")
 TARGET_LINK_LIBRARIES(_gsp_sc_sa.so ${SystemC_LIBRARIES}
