diff -Naur grlib-orig/lib/gaisler/misc/gptimer.vhd grlib/lib/gaisler/misc/gptimer.vhd
--- grlib-orig/lib/gaisler/misc/gptimer.vhd	2009-09-04 10:20:55.000000000 +0200
+++ /grlib/lib/gaisler/misc/gptimer.vhd	2010-04-26 15:48:12.000000000 +0200
@@ -45,7 +45,7 @@
     pirq     : integer := 0;
     sepirq   : integer := 0;	-- use separate interrupts for each timer
     sbits    : integer := 16;			-- scaler bits
-    ntimers  : integer range 1 to 7 := 1; 	-- number of timers
+    ntimers  : integer := 1; 	-- number of timers
     nbits    : integer := 32;			-- timer bits
     wdog     : integer := 0
   );
@@ -274,11 +274,11 @@
 -- boot message
 
 -- pragma translate_off
-    bootmsg : report_version 
-    generic map ("gptimer" & tost(pindex) & 
-	": GR Timer Unit rev " & tost(REVISION) & 
-	", " & tost(sbits) & "-bit scaler, " & tost(ntimers) & 
-	" " & tost(nbits) & "-bit timers" & ", irq " & tost(pirq));
+--    bootmsg : report_version 
+--    generic map ("gptimer" & tost(pindex) & 
+--	": GR Timer Unit rev " & tost(REVISION) & 
+--	", " & tost(sbits) & "-bit scaler, " & tost(ntimers) & 
+--	" " & tost(nbits) & "-bit timers" & ", irq " & tost(pirq));
 -- pragma translate_on
 
 end;
